src/main/java/org/generationcp/middleware/service/DataImportServiceImpl.java

```diff
@@ -368,6 +392,57 @@ public void assignLocationIdVariableToEnvironmentDetailSection(final Workbook wo
 
 	}
 
+	@Override
+	public void processExperimentalDesign(final Workbook workbook, final String programUUID, final String exptDesignValueFromObsSheet)
+		throws WorkbookParserException {
+		final Optional<MeasurementVariable> exptDesignVarInConditions =
+			this.findMeasurementVariableByTermId(TermId.EXPERIMENT_DESIGN_FACTOR.getId(), workbook.getConditions());
+		if(exptDesignVarInConditions.isPresent()) {
+			final MeasurementVariable exptDesignVar = exptDesignVarInConditions.get();
+			if (exptDesignVar.getValue() != null && !exptDesignVar.getValue().isEmpty()) {
+				Term exptDesignValue = this.termDataManager.getTermByName(exptDesignVar.getValue());
+				if (exptDesignValue != null) {
+					exptDesignVar.setValue(String.valueOf(exptDesignValue.getId()));
+				} else {
+					throw new WorkbookParserException("Invalid value for Experimental Design");
+				}
+			} else {
+				exptDesignVar.setValue(this.getExperimentalDesignIdValue(exptDesignValueFromObsSheet));
+			}
+		}
+
+		final Optional<MeasurementVariable> exptDesignVarInFactors =
+			this.findMeasurementVariableByTermId(TermId.EXPERIMENT_DESIGN_FACTOR.getId(), workbook.getFactors());
+		if(exptDesignVarInFactors.isPresent()) {
+			final MeasurementVariable exptDesignVar = exptDesignVarInFactors.get();
```

Comment:
Is there a case where exptDesignVarInFactors.get() is not present?

Reply:
Yes. When the Expt_design is not present in the import file.